stairctl/src/main/java/bio/terra/stairctl/StairwayService.java

```diff
@@ -0,0 +1,93 @@
+package bio.terra.stairctl;
+
+import bio.terra.stairctl.configuration.StairwayConfiguration;
+import bio.terra.stairway.Control;
+import bio.terra.stairway.Stairway;
+import java.util.Properties;
+import javax.sql.DataSource;
+import org.apache.commons.dbcp2.ConnectionFactory;
+import org.apache.commons.dbcp2.DriverManagerConnectionFactory;
+import org.apache.commons.dbcp2.PoolableConnection;
+import org.apache.commons.dbcp2.PoolableConnectionFactory;
+import org.apache.commons.dbcp2.PoolingDataSource;
+import org.apache.commons.pool2.ObjectPool;
+import org.apache.commons.pool2.impl.GenericObjectPool;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class StairwayService {
+  private static final Logger logger = LoggerFactory.getLogger(StairwayService.class);
+
+  private final StairwayConfiguration stairwayConfiguration;
+  private Stairway stairway;
+  private Control control;
+  private ConnectParams currentConnectParams;
+
+  @Autowired
+  public StairwayService(StairwayConfiguration stairwayConfiguration) {
+    this.stairwayConfiguration = stairwayConfiguration;
+  }
+
+  public void connectStairway(ConnectParams connectParams) {
+    // Fill in the defaults from the configuration
+    connectParams.applyDefaults(stairwayConfiguration.makeConnectParams());
+    DataSource dataSource = configureDataSource(connectParams);
+
+    try {
+      this.stairway =
+          Stairway.newBuilder()
+              .stairwayName("stairctl") // No queue, no flights, so name is irrelevant...
+              .stairwayClusterName("stairctlcluster") // ...as is clustername
+              .enableWorkQueue(false) // we will not be running any flights anywhere!
+              .build();
+
+      stairway.initialize(dataSource, false, false);
```

Comment:
might be nice to comment about how this is explicitly *not* trying to start the stairway instance and why.